### ISSUE TITLE
[Bug]: Crash if fullscreen mode not enabled

### ISSUE THREAD

**Issue by omaSpencer** [tsparticles/tsparticles#4228]
### Contact Details

oma.spencer@gmail.com

### What happened?

I tried disable fullscreen mode, but if I set to false, then works with fully lagged and crashed out. Particles disappear and showing a sad emoji. Memory leak + chrome tab crash the response.

Component

```
import { Component } from 'solid-js';
import TSParticles from 'solid-particles';
import { loadFull } from 'tsparticles';
import './styles.scss';

const Particles: Component = () => {
  const particlesInit = async (main) => {
    await loadFull(main);
  };

  return (
    <>
      <TSParticles
        className='particles'
        init={particlesInit}
        options={{
          fullScreen: {
            enable: false,
          },
          fpsLimit: 60,
          interactivity: {
            events: {
              resize: true,
            },
            modes: {
              bubble: {
                distance: 400,
                duration: 2,
                opacity: 0.8,
                size: 40,
              },
              push: {
                quantity: 4,
              },
              repulse: {
                distance: 200,
                duration: 0.4,
              },
            },
          },
          particles: {
            color: {
              value: '#ffffff',
            },
            links: {
              color: '#ffffff',
              distance: 150,
              enable: true,
              opacity: 0.5,
              width: 1,
            },
            collisions: {
              enable: true,
            },
            move: {
              direction: 'none',
              enable: true,
              outModes: {
                default: 'bounce',
              },
              random: false,
              speed: 6,
              straight: false,
            },
            number: {
              density: {
                enable: true,
                area: 800,
              },
              value: 80,
            },
            opacity: {
              value: 0.5,
            },
            shape: {
              type: 'circle',
            },
            size: {
              value: { min: 1, max: 5 },
            },
          },
          detectRetina: true,
        }}
      />
    </>
  );
};

export default Particles;
```

### tsParticles Version

2.0.6

### Which library are you using?

Solid (solid-particles)

### tsParticles Configuration

```json
{
          fullScreen: {
            enable: false,
          },
          fpsLimit: 60,
          interactivity: {
            events: {
              resize: true,
            },
            modes: {
              bubble: {
                distance: 400,
                duration: 2,
                opacity: 0.8,
                size: 40,
              },
              push: {
                quantity: 4,
              },
              repulse: {
                distance: 200,
                duration: 0.4,
              },
            },
          },
          particles: {
            color: {
              value: '#ffffff',
            },
            links: {
              color: '#ffffff',
              distance: 150,
              enable: true,
              opacity: 0.5,
              width: 1,
            },
            collisions: {
              enable: true,
            },
            move: {
              direction: 'none',
              enable: true,
              outModes: {
                default: 'bounce',
              },
              random: false,
              speed: 6,
              straight: false,
            },
            number: {
              density: {
                enable: true,
                area: 800,
              },
              value: 80,
            },
            opacity: {
              value: 0.5,
            },
            shape: {
              type: 'circle',
            },
            size: {
              value: { min: 1, max: 5 },
            },
          },
          detectRetina: true,
        }
```

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by matteobruni** [comment]
Thank you for the report, I never noticed this issue. A fix will be released in the 2.1.0 version, that can't be released without it. #4031 for reference

https://codesandbox.io/s/amazing-christian-epc49y?file=/src/particles.jsx reproducible sample

**Comment by matteobruni** [comment]
I've seen that this issue is replicable also in the vanilla library, if a canvas is placed inside the containing div (like the one used by `solid-particles`, the issue is the same.

The canvas starts resizing infinitely only with a pixel ratio != 1 (like retina displays)
